src/ModelContextProtocol/Server/McpServerExtensions.cs

```diff
@@ -169,6 +169,26 @@ public static Task<ListRootsResult> RequestRootsAsync(
             cancellationToken);
     }
 
+    /// <summary>
+    /// Requests the client to list the roots it exposes.
+    /// </summary>
+    /// <param name="server">The server issueing the request.</param>
+    /// <param name="notification">The notification to send.</param>
+    /// <param name="cancellationToken">A token to cancel the operation.</param>
+    /// <returns>A task containing the response from the client.</returns>
+    public static Task NotifyProgressAsync(
+        this IMcpServer server,
+        ProgressNotificationValue notification,
```

Comment:
Doesn't this need to be a ProgressNotification rather than a ProgressNotificationValue?